### PR TITLE
Create Dynamic Array.cpp

### PR DESCRIPTION
**Pull request by nitinshukla413**

### CODE REVIEW

**Comment by nitinshukla413** [comment]
Can you please add Hacktober label to this?

File: HackerRank/Dynamic Array.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#include <iostream>
```
Comment (amanBhawsar) [inline HackerRank/Dynamic Array.cpp:1]: add problem link here

File: HackerRank/Dynamic Array.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+template<class T> inline T abs(T a){return a > 0 ? a : -a;}
+
+typedef long long ll;
+typedef pair<int, int> ii;
+typedef vector<int> vi;
+
+const int inf = 1e9 + 143;
+const ll longinf = 1e18 + 143;
+
+inline int read(){int x;scanf(" %d",&x);return x;}
```
Comment (amanBhawsar) [inline HackerRank/Dynamic Array.cpp:25]: don't use template, write simple solution

File: HackerRank/Dynamic Array.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#ifdef KAZAR
+  freopen("f.input","r",stdin);
+  freopen("f.output","w",stdout);
+  freopen("error","w",stderr);
+#endif
+
+  int n = read();
+  int q = read();
+  assert(1 <= n && n <= 1e5);
+  assert(1 <= q && q <= 1e5);
```
Comment (amanBhawsar) [inline HackerRank/Dynamic Array.cpp:38]: Do it without assert

File: HackerRank/Dynamic Array.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+inline int read(){int x;scanf(" %d",&x);return x;}
+
+int main(){
+
+#ifdef KAZAR
+  freopen("f.input","r",stdin);
+  freopen("f.output","w",stdout);
+  freopen("error","w",stderr);
+#endif
```
Comment (amanBhawsar) [inline HackerRank/Dynamic Array.cpp:33]: remove this

File: HackerRank/Dynamic Array.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  int n = read();
+  int q = read();
+  assert(1 <= n && n <= 1e5);
+  assert(1 <= q && q <= 1e5);
+  int lastans = 0;
+  vector<vi> seq(n);
+  for (int i = 0; i < q; i++) {
```
Comment (amanBhawsar) [inline HackerRank/Dynamic Array.cpp:41]: add 2-3 line comment for logic understanding so that even beginner can get it.

**Review by amanBhawsar — CHANGES_REQUESTED** [review]

**Comment by amanBhawsar** [comment]
@nitinshukla413  Hey I have added some comments please check them. Also, solve 5+ Questions and then raise PR.

**Pull request closed without merging**